### ISSUE TITLE
payfastRequest.ToString() doesn't emit the custom_str properties

### ISSUE THREAD

**Issue by ringwood-dsg** [louislewis2/payfast#24]
In my request I am assigning a specific value to the `custom_str1` field. When you call the `.ToString()` extension, it seems its not emitting that information in the final URI or signature.

The response from the sandbox doesn't include this value.

**Comment by TechnoDeziPtyLtd** [comment]
Hi

Which function are you using? Or do you have a code example?
I've used it a couple of times but haven't specifically checked the response from sandbox. I can test it tomorrow, just need to know which function you are using. Notmal payment, subscription, etc?
Also keep in mind special characters and trailing spaces for Payfast is an issue, so make sure you filter them out.

**Comment by ringwood-dsg** [comment]
I'm building a `PayFastRequest` like so:

```
var subsRequest = new PayFastRequest(_payFast.PassPhrase);
subsRequest.merchant_id = _payFast.MerchantId;
subsRequest.merchant_key = _payFast.MerchantKey;
//add other details as required

//Add custom value.
subsRequest.custom_str1 = "1234"

string payfastUri = $"{_payFast.ProcessUrl}{subsRequest.ToString()}";
return await Task.FromResult(new JsonResult(payfastUri)); //Set breakpoint here and see that <payfastUri> doesn't include custom_str1
```

The problem seems to be in the `.ToString()` extension that doesn't include the `custom_str1` property for serialization.

**Comment by TechnoDeziPtyLtd** [comment]
Thanks for the code, I'll test it tomorrow and let you know. If it is a problem I'll push through a code update

**Comment by ringwood-dsg** [comment]
Just on that note, I realised that I might need both `custom_str1` and `custom_str2` and gave that combination a try. So I would add to the request:

```
subsRequest.custom_str1 = "customvalue1";
subsRequest.custom_str2 = "customvalue2";
```

I debugged it again to see if there's any change and now I see that the `custom_str2` is in the URI request, but **not** `custom_str1`. 

Hope that helps. Your time and effort is greatly appreciated!
